pkg/plugin/processor/standalone/wasm_processor.go

```diff
@@ -0,0 +1,258 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package standalone
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit-processor-sdk/serde"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/rs/zerolog"
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+)
+
+type wasmProcessor struct {
+	logger zerolog.Logger
+
+	runtime       wazero.Runtime
+	runtimeCancel context.CancelFunc
+
+	// commandCh is used to send commands to
+	// the actual processor (the WASM module)
+	// and through exported host functions.
+	commandCh chan sdk.Command
+	// replyErr is used to communicate replies between
+	// the actual processor (the WASM module) and wasmProcessor
+	replyCh chan sdk.CommandResponse
+	// replyErr is used to communicate errors between
+	// the actual processor (the WASM module) and wasmProcessor
+	replyErr chan error
+	// runModStopped is used to know when the run module stopped
+	runModStopped chan struct{}
+}
+
+func NewWASMProcessor(ctx context.Context, logger zerolog.Logger, wasmPath string) (sdk.Processor, error) {
+	runtimeCtx, runtimeCancel := context.WithCancel(ctx)
+	r := wazero.NewRuntimeWithConfig(
+		runtimeCtx,
+		wazero.NewRuntimeConfig().WithCloseOnContextDone(true),
+	)
+	wasi_snapshot_preview1.MustInstantiate(ctx, r)
+
+	p := &wasmProcessor{
+		logger:        logger,
+		runtime:       r,
+		runtimeCancel: runtimeCancel,
+		commandCh:     make(chan sdk.Command),
+		replyCh:       make(chan sdk.CommandResponse),
+		replyErr:      make(chan error),
```

Comment:
Nice, I like the way communication works through these channels and it's abstracted away from the caller. I'm just wondering if it would be simpler to have a single channel for the reply, maybe by having a struct that contains either a `sdk.CommandResponse` or an `error`.